research/2019-12-16_searching_with_the_calm_api.md

```diff
@@ -0,0 +1,241 @@
+# Searching with the CALM API
+
+### 16 December 2019, AWLC
+
+This has some notes on using the CALM API, and an example script for searching all the records in the CALM catalogue.
+
+
+
+## Motivation
+
+As part of the born-digital content migration, I had a collection of directories with accession-like numbers, and I wanted to get a list of all the accession numbers in CALM, so I could match the directories to accession numbers:
+
+    AAUICW_2185 ~> AAUICW/2185 ?
+    AAUNAM_2213 ~> AAUNAM/2213 ?
+    AAUNAM_2254 ~> AAUNAM/2254 ?
+
+Exporting a list of accession numbers from the CALM GUI would have been tricky (I don't have access to CALM on my Mac), so I tried using the CALM API to export the data.
+
+<details>
+  <summary>What is an accession/accession number?</summary>
+
+  > In library science, an *accession* is all the material that gets collected at one time from a single source.
+  > The *accession identifier* is the unique identifier given to that batch of material.
+  >
+  > At Wellcome, the accession identifier is made of two parts: the *accession number* (sequential, currently up to 2551) and a *collection reference* (such as WT, PPMTW, SANCT).
+  > The two are written together, separated by a slash.
+  > Examples:
+  >
+  > The accession number is globally unique: if there's an accession `WT/123`, you won't see `123` with any other collection reference.
```

Comment:
Hmm, is this how Trudy expressed them in the spreadsheet? There a slight danger of accidentally duplicating catalogue references for early accessions and I wonder whether it would be sensible to adopt some other separator than the slash here to avoid confusion later. e.g. if Acc_No 1 was a WT accession, WT/1 could refer to _either_ WT accession 1 _or_ WT item 1 in the catalogue.

Reply:
Not sure, this might be a convention that I've accidentally invented! I think I saw what looked like `WT/123` hanging around in Preservica, which is where I got the idea. I can’t find anything from Trudy that specifically uses this convention. Definitely see the point about ambiguity.

I don't think Archivematica or the Wellcome Storage mind what you put here, so we can choose a different character. Free choice!

The V drive uses underscores as separators. We could do that.

If not, maybe a hash `#`, because that's commonly used to mean “number”? e.g. `WT#1` or even `WT#acc1`.

Your team will have to deal with these identifiers, so happy to take your lead on this one.